homeassistant/components/esphome/entity.py

```diff
@@ -215,24 +216,6 @@ def __init__(
         self._attr_device_info = DeviceInfo(
             connections={(dr.CONNECTION_NETWORK_MAC, device_info.mac_address)}
         )
-        #
-        # If `friendly_name` is set, we use the Friendly naming rules, if
-        # `friendly_name` is not set we make an exception to the naming rules for
-        # backwards compatibility and use the Legacy naming rules.
-        #
-        # Friendly naming
-        # - Friendly name is prepended to entity names
-        # - Device Name is prepended to entity ids
-        # - Entity id is constructed from device name and object id
-        #
-        # Legacy naming
-        # - Device name is not prepended to entity names
-        # - Device name is not prepended to entity ids
-        # - Entity id is constructed from entity name
-        #
-        if not device_info.friendly_name:
```

Comment:
Should we print a warning to the user that they should update their ESPHome YAML ?

Reply:
Ideally, yes, however, I think that'll be contentious, since some users have actively said they don't want to add a friendly name in the past. I don't know how large that segment of users is, but I'm sure we'll get complaints if we do that.

Reply:
I can imagine some folks might point out that `friendly_name` is listed as optional in the [ESPHome docs](https://esphome.io/components/esphome.html#configuration-variables).

Maybe a good middle ground could be updating the docs to say it's *highly recommended*, and then logging an info message here when it's not set? That way we're nudging folks in the right direction without being too strict.

Reply:
Adding a test as well

Reply:
Test added